src/bindings/scala/api/src/main/scala/com/ctc/omega_edit/api/Session.scala

```diff
@@ -0,0 +1,41 @@
+/*
+ * Copyright 2021 Concurrent Technologies Corporation
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.ctc.omega_edit.api
+
+import com.ctc.omega_edit.api.Change.Result
+
+/**
+  * The top level type in OmegaEdit, maintains the data and all instances related to it.
+  * Provides mutators and viewport factory methods.
+  */
+trait Session {
+  def size: Long
+  def isEmpty: Boolean
+
+  def push(s: String): Result
+  def push(b: Array[Byte]): Result
```

Comment:
For push, it's not clear from the name if these are pushing data to the front or to the back.  If this is a convenience for streaming, then may I suggest we use append as the name to make it clear where the data is being inserted?